src/Tickets/Emails/Provider.php

```diff
@@ -37,15 +37,18 @@ public function register() {
 		// Register singletons.
 		$this->container->singleton( static::class, $this );
 		$this->container->singleton( 'tickets.emails.provider', $this );
+
 		$this->container->singleton( Admin\Emails_Tab::class );
-		$this->container->singleton( 'tickets.emails.emails-tab', $this );
+		$this->container->singleton( 'tickets.emails.emails-tab', Admin\Emails_Tab::class );
 
 		$this->container->singleton( Admin\Preview_Modal::class );
-		$this->container->singleton( 'tickets.emails.admin.preview-modal', $this );
+		$this->container->singleton( 'tickets.emails.admin.preview-modal', Admin\Preview_Modal::class );
+
+		$this->container->register( Email_Handler::class );
+		$this->container->singleton( 'tickets.emails.email-handler', Email_Handler::class );
```

Comment:
We dont need the string based naming anymore. You can just use the `Email_Class::class` one.